### ISSUE TITLE
Explain This Program to Me Like I Am Stupid, Because I Am

### ISSUE THREAD

**Issue by TurtleShroom** [haifengkao/StellairsLoadOrderFixer24#5]
Good evening.

Okay. I downloaded the alphabetization sorting file for the Launcher, put it in the My Documents' "Stellaris" root folder, and ran it. I double clicked it and the Command Prompt appeared for a few seconds. 

My JSON files were noted to have been modified. Two new BAK files were present alongside the two files allegedly overwritten. I then opened the Launcher. Nothing happened. My files were still totally out of order.

What did I do wrong? Do I need to install the Python system on my computer? I didn't think I needed to because of the Command Prompt and the successful overwriting.

I am an idiot who can't code my way out of a paper bag. I am humbly asking the create of this alphabetization script to explain how to install, use, and fix his program as if I were a moronic five year-old.

I am suffering withdrawal. I need my "Stellaris" fix.

Thank you for any assistance you may render.
-TURTLESHROOM

P.S.: I didn't see a pop-up that read "DONE!".

**Comment by haifengkao** [comment]
Some people report the same issue. Could you run the program from the command line?
It should print out the error message. 
If it shows xxx key not found, please give me the content of the following files:
```
mods_registry.json
dlc_load.json
game_data.json
````

**Comment by TurtleShroom** [comment]
Yes. Double clicking the program made the Command Prompt pop up for a few seconds. I ran it as an Administrator. I'll get you the JSON files shortly.

**Comment by TurtleShroom** [comment]
Here are my JSON files: [JSON.zip](https://github.com/haifengkao/StellairsLoadOrderFixer24/files/3725473/JSON.zip).

**Comment by TurtleShroom** [comment]
I should note that my My Documents folder is in the Registry and has been assigned non-default zone for nearly a decade. Someone told me to try downloading the Python and running the PY file itself.

![Untitled](https://user-images.githubusercontent.com/17663470/66778675-68ac3000-ee9a-11e9-832a-e0f7e502debf.png)

It failed.

**Comment by TurtleShroom** [comment]
I tried editing the file like I was told, fitting in my custom path.

This is the path: _I:\Documents and Settings\[NAME]\My Documents_ . The Paradox and "Stellaris" folders are as you would expect.

I changed the script to read this:
`locations = os.path.join('I:\\', 'Documents and Settings', 'tyler', 'My Documents', 'Paradox Interactive', 'Stellaris'), ".", "..", os.path.join(
    os.path.expanduser('~'), '.local', 'share', 'Paradox Interactive', 'Stellaris')]`

From this:
`locations = [os.path.join(os.path.expanduser('~'), 'Documents', 'Paradox Interactive', 'Stellaris'), ".", "..", os.path.join(
    os.path.expanduser('~'), '.local', 'share', 'Paradox Interactive', 'Stellaris')]`

!https://media.discordapp.net/attachments/488818218841145344/633387379092357131/Untitled.png

I failed.

**Comment by TurtleShroom** [comment]
![image](https://user-images.githubusercontent.com/17663470/66780040-d574f980-ee9d-11e9-8dba-505511f473f6.png)

**Comment by TurtleShroom** [comment]
![image](https://user-images.githubusercontent.com/17663470/66780592-33eea780-ee9f-11e9-8f66-771c59a04299.png)

**Comment by TurtleShroom** [comment]
![image](https://user-images.githubusercontent.com/17663470/66782541-e7f23180-eea3-11e9-8217-c88263e1e277.png)

![image](https://user-images.githubusercontent.com/17663470/66782700-4a4b3200-eea4-11e9-9f53-7e4a3eca7def.png)

^ This is what I got with the new version released today.

**Comment by TurtleShroom** [comment]
Running it on my desktop:

![image](https://user-images.githubusercontent.com/17663470/66782769-78c90d00-eea4-11e9-95c3-0c0674d7a2c8.png)

**Comment by haifengkao** [comment]
I have updated the python script (not exe, because I don't have windows at the moment)
Please run it from the mod folder. It will force python using utf8 encoding.

**Comment by TurtleShroom** [comment]
I put it in the "MOD" folder like you said.

![image](https://user-images.githubusercontent.com/17663470/66849218-88982e00-ef44-11e9-8f27-c6dccccce166.png)
^ This is what I got.

What is the default path your script assumes the files are in? I can always create that directory in a string of empty folders and move it there.

**Comment by TurtleShroom** [comment]
Is the "CP1252.PY" file corrupted? I downloaded the Python from the official Python site, which is said to be V3.7.4 . It is a sixty-four bit installation placed in the sixty-four bit Program Files directory. PATH is on the installation EXE and all the official options were checked.

I am going to uninstall the Python and then reinstall it, and see if that changes anything.

**Comment by haifengkao** [comment]
I know where the problem is. Don't use python3, use python2 instead

**Comment by TurtleShroom** [comment]
I will attempt that now. Thank you.

**Comment by TurtleShroom** [comment]
What version of Python II do I need? Do I need V2.6.7 ?

**Comment by haifengkao** [comment]
I use python 2.7.16

**Comment by TurtleShroom** [comment]
Will V2.7.15 work? I couldn't find the one you mentioned.

**Comment by haifengkao** [comment]
It's on the official site
https://www.python.org/downloads/release/python-2716/

**Comment by TurtleShroom** [comment]
**IT WORKED!!**

![image](https://user-images.githubusercontent.com/17663470/66853498-69ea6500-ef4d-11e9-895e-3322baf3bc96.png)

Thank you so much! Make sure that you specify that you need Python _**II**_ and not _III_.

You told me that it was to be alphabetized in descending order, that is, in reverse. Is this intended? I was told that Paradox flipped it back in V2.4.1 , as opposed to V2.4 .
